### PR TITLE
Fix build for paths with spaces

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to https://github.com/dotnet/corefx/pull/34315

Fix build errors when paths contain empty spaces in the init-tools and optional tools setup.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/OptionalTooling.targets [hunk 1]
```diff
@@ -33,15 +33,15 @@
   -->
   <Target Name="GetOptionalToolingPaths">
     <PropertyGroup>
-      <OptionalToolingDir>$(ToolsDir)optional-tool-runtime\</OptionalToolingDir>
+      <OptionalToolingDir>$(ToolsDir)optional-tool-runtime</OptionalToolingDir>
```
Comment (stephentoub) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/OptionalTooling.targets]: Why did you move the `\` from here to all of the places it's used?

Comment (ViktorHofer) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/OptionalTooling.targets]: Because a quote follows the backslash further down which causes parsing errors: https://github.com/dotnet/buildtools/pull/2212/files#diff-27ba4b34f549889e007e06d384e699ecR85

Comment (stephentoub) [reply src/Microsoft.DotNet.Build.Tasks/PackageFiles/OptionalTooling.targets]: I see.

**Comment by ViktorHofer** [comment]
Thanks!
